third_party/sigs.k8s.io/controller-runtime/pkg/dynamiccache/cache.go

```diff
@@ -67,3 +72,15 @@ func defaultOpts(config *rest.Config, opts cache.Options) (cache.Options, error)
 	}
 	return opts, nil
 }
+
+func convertToSelectorsByGVK(selectorsByObject cache.SelectorsByObject, scheme *runtime.Scheme) (internal.SelectorsByGVK, error) {
+	selectorsByGVK := internal.SelectorsByGVK{}
+	for object, selector := range selectorsByObject {
+		gvk, err := apiutil.GVKForObject(object, scheme)
+		if err != nil {
+			return nil, err
+		}
+		selectorsByGVK[gvk] = selector
```

Comment:
@shomron I am getting error

```
cannot use selector (variable of type internal.Selector) as internal.Selector value in assignmentcompilerIncompatibleAssign
```

any ideas?

Reply:
Opened https://github.com/sozercan/gatekeeper/pull/73 with fixes to the rebase.

Reply:
thanks @shomron!